symbolic-base/src/ZkFold/Symbolic/Algorithms/ECDSA/ECDSA.hs

```diff
@@ -1,51 +1,53 @@
 {-# LANGUAGE AllowAmbiguousTypes #-}
 {-# LANGUAGE RebindableSyntax    #-}
 {-# LANGUAGE TypeOperators       #-}
+
 module ZkFold.Symbolic.Algorithms.ECDSA.ECDSA where
 import           Data.Type.Equality
-import           GHC.Base                                (($))
-import           GHC.TypeLits                            (KnownNat, Log2)
-import qualified GHC.TypeNats
+import           GHC.TypeLits                            (KnownNat)
 
 import           ZkFold.Base.Algebra.Basic.Class         hiding (Euclidean (..))
-import           ZkFold.Base.Algebra.Basic.Number        (value)
 import           ZkFold.Base.Algebra.EllipticCurve.Class
 import qualified ZkFold.Symbolic.Class                   as S
 import           ZkFold.Symbolic.Data.Bool
-import           ZkFold.Symbolic.Data.ByteString         (ByteString)
-import           ZkFold.Symbolic.Data.Combinators        (Iso (..), NumberOfRegisters, RegisterSize (Auto))
+import           ZkFold.Symbolic.Data.Combinators        (GetRegisterSize, NumberOfRegisters, RegisterSize (Auto))
 import           ZkFold.Symbolic.Data.Conditional
 import           ZkFold.Symbolic.Data.Eq
-import           ZkFold.Symbolic.Data.FieldElement       (FieldElement)
-import           ZkFold.Symbolic.Data.UInt               (UInt, eea)
+import           ZkFold.Symbolic.Data.FFA                (FFA, KnownFFA, toUInt)
+import           ZkFold.Symbolic.Data.UInt               (UInt)
 
+-- TODO: verify the actual message instead of a hash
 ecdsaVerify
-  :: forall point n c curve baseField.
-     ( S.Symbolic c
-     , KnownNat n
-     , baseField ~ UInt 256 'Auto c
-     , ScalarFieldOf point ~ FieldElement c
+  :: forall n point curve p q baseField scalarField ctx .
+     ( S.Symbolic ctx
+     , baseField ~ FFA q 'Auto ctx
+     , scalarField ~ FFA p 'Auto ctx
      , point ~ Weierstrass curve (Point baseField)
+     , ScalarFieldOf point ~ scalarField
      , CyclicGroup point
```

Comment:
Is this constraint necessary?

Reply:
Nono, I'm talking about `CyclicGroup point` constraint
Isn't it satisfied due to all the other constraints?